app/content-panel/blogs/[id]/page.tsx

```diff
@@ -0,0 +1,157 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import styles from './BlogDetailsPage.module.scss';
+import BlogDetailsCard from '../../../../components/reusable-components/blogDetails-card/BlogDetailsCard';
+
+interface Author {
+  first_name: string;
+  last_name: string;
+}
+interface BlogDetails {
+  title: string;
+  content: string;
+  image: string;
+  author: Author;
+  publishDate: string;
+  tags: string[];
+}
+
+const BlogDetailsPage = ({ params }: { params: { id: string } }) => {
+  const [isEditable, setIsEditable] = useState(false);
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const [blogDetails, setBlogDetails] = useState<BlogDetails>({
+    title: 'N/A',
+    content: 'N/A',
+    image: '',
+    author: { first_name: 'N/A', last_name: 'N/A' },
+    publishDate: 'N/A',
+    tags: [],
+  });
+
+  const [uploadError, setUploadError] = useState<string | null>(null);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          `${process.env.NEXT_PUBLIC_API_BASE_URL}/blog-posts/${params.id}`
+        );
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        const data = await response.json();
```

Comment:
please look into api/queries. There you can find how we are fetching data for this project. We are using react-query.
Remove this block of code and create another file there, or see if you can use api/queries/blogs/getBlogs.ts